lib/dry/matcher/match.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'dry/matcher'
+
+module Dry
+  class Matcher
+    Match = ::Proc.new do |value, patterns|
```

Comment:
Perhaps this could be called `PatternMatch`, to make its purpose clearer?

Reply:
Sure, I'll change it.

Reply:
Done.